DEVELOPMENT.md

```diff
@@ -2,19 +2,50 @@
 
-    1. create a `requirements.txt` file in the specific example's folder, and add the notebook's dependency specifications there
+    > eg. the [find_label_errors_iris](find_label_errors_iris/find_label_errors_iris.ipynb) notebook will have a relative path of `find_label_errors_iris/find_label_errors_iris.ipynb` and will have the badge:
+    > 
+    > ```
+    > [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/cleanlab/examples/blob/master/find_label_errors_iris/find_label_errors_iris.ipynb.ipynb)
+    > ```
+
+    This will create a badge that will link to a Google Colab version of the notebook. 
+    
+    Note that the Colab badge links to the notebook in the master branch, so at the time of making the PR, the link will be invalid. Please remember to check that the Colab link works after the PR has been approved and merged to `master`.
+
+- Use `pip freeze` to determine the package versions that are used, then
+
+    1. create a `requirements.txt` file in the specific example's folder, and add the notebook's dependency specifications there.
+    
+    2. add the dependency specifications in the main `requirements.txt` file, ensuring that the new dependencies do not conflict with the existing ones.
+
+    3. add a markdown block right above the code cell that imports the dependencies that reads:
     
-    2. add the dependency specifications in the main `requirements.txt` file, ensuring that the new dependencies do not conflict with the existing ones
+        > Please install the dependencies specified in this requirements.txt file before running the notebook.
+
+        where the requirements.txt text hyperlinks to the `requirements.txt` file located in that specific example's folder (the file created in step 1). Be sure to use the absolute github link (instead of relative path) as Google Colab will have no access to other files. 
+
+        > eg. for the [find_label_errors_iris](find_label_errors_iris/find_label_errors_iris.ipynb) notebook, we would hyperlink the following url:
+        > 
+        > ```
+        > https://github.com/cleanlab/examples/blob/master/find_label_errors_iris/requirements.txt
+        > ```
+
+- If the notebook takes a long time to run or is hard to auto-execute, add its folder name to the `ignore_folders` list in [run_all_notebooks.py](run_all_notebooks.py).
 
 - Add the notebook to the [Table of Contents](https://github.com/cleanlab/examples#table-of-contents)
- table in the README, ideally grouping the newly added example with any other related examples
+ table in the README, ideally grouping the newly added example with any other related examples.
+
+- After a new notebook has been added and pushed to `master` branch, refrain from changing the notebook and folder names, as many links referencing the example notebook will be used throughout cleanlab documentation, blog posts and more.
```

Comment:
```suggestion
- After a new notebook has been added and pushed to `master` branch, AVOID changing the notebook and folder names, as this may  break existing links referencing the example notebook throughout cleanlab documentation, blog posts, and more.
```